sdk/attestation/azure-security-attestation/azure/security/attestation/_client.py

```diff
@@ -199,57 +299,112 @@ def attest_open_enclave(self, report, inittime_data=None, runtime_data=None, **k
+        # Note that this must be done before calling into the implementation
+        # layer because the implementation layer doesn't like keyword args that
+        # it doesn't expect :(.
+        options = merge_validation_args(
+            self._config._kwargs, kwargs  # pylint: disable = protected-access
+        )
+
         result = self._client.attestation.attest_open_enclave(request, **kwargs)
-        token = AttestationToken[GeneratedAttestationResult](token=result.token,
-            body_type=GeneratedAttestationResult)
-        token.validate_token(self._config.token_validation_options, self._get_signers(**kwargs))
-        return AttestationResponse[AttestationResult](token, AttestationResult._from_generated(token.get_body()))
+        token = AttestationToken(
+            token=result.token, body_type=GeneratedAttestationResult
+        )
+
+        if options.get("validate_token", True):
+            token._validate_token(  # pylint: disable = protected-access
+                self._get_signers(**kwargs), **options
+            )
+        return (
+            AttestationResult._from_generated(  # pylint: disable = protected-access
+                token._get_body()  # pylint: disable = protected-access
+            ),
+            token,
+        )  # pylint: disable = protected-access
 
     @distributed_trace
-    def attest_tpm(self, request, **kwargs):
-        #type:(TpmAttestationRequest, **Any) -> TpmAttestationResponse
-        """ Attest a TPM based enclave.
+    def attest_tpm(self, content, **kwargs):
+        # type: (str, **Any) -> str
+        """Attest a TPM based enclave.
 
-        See the `TPM Attestation Protocol Reference <https://docs.microsoft.com/en-us/azure/attestation/virtualization-based-security-protocol>`_ for more information.
+        See the `TPM Attestation Protocol Reference
+        <https://docs.microsoft.com/en-us/azure/attestation/virtualization-based-security-protocol>`_
+        for more information.
 
-        :param request: Incoming request to send to the TPM attestation service.
-        :type request: azure.security.attestation.TpmAttestationRequest
+        :param str content: Data to send to the TPM attestation service.
         :returns: A structure containing the response from the TPM attestation.
-        :rtype: azure.security.attestation.TpmAttestationResponse
+        :rtype: str
         """
-        response = self._client.attestation.attest_tpm(request.data, **kwargs)
-        return TpmAttestationResponse(response.data)
+
+        response = self._client.attestation.attest_tpm(
+            content.encode("ascii"), **kwargs
+        )
+        return response.data.decode("ascii")
 
     def _get_signers(self, **kwargs):
-        # type:(Any) -> list[AttestationSigner]
-        """ Returns the set of signing certificates used to sign attestation tokens.
-        """
+        # type: (Any) -> list[AttestationSigner]
```

Comment:
nit: also not a big deal since this is a private method
```suggestion
        # type: (**Any) -> list[AttestationSigner]
```